### DISCUSSION TITLE
Add support for DecisionMatrixDefinitionVersion and ExpressionSetDefinitionVersion

### DISCUSSION THREAD

**Discussion by twardegaSFDC** [forcedotcom/cli#2850] [Ideas]
We are using DecisionMatirixDefinitions and ExpressionSetDefinitions in our PSS project on GovCloud+. We are using SF CLI v 2.38.7.
We deployed DecisionMatrixDefinitions and ExpressionSetDefinitions to sandboxes and production (go-live a week ago) without issues.
We need to deploy new versions of DecisonMatrixDefinitions and ExpressionSetDefinitions and it looks that DecisionMatrixDefinitionVersion and ExpressionSetDefinitionVersion metadata is not supported by Salesforce CLI at this time:
- attempt to retrieve a specific version using DecisonMatrixDefinitionVersion and ExpressionSetDefinitionVersion does not pull the corresponding metadata from org to a corresponding folder
- attemps to deploy a specific version using DecisonMatrixDefinitionVersion and ExpressionSetDefinitionVersion fails with error 'No sorce-backed components present in the package' (even if the folder and files are created by hand)

**Comment by VivekMChawla** [comment]
@tomekwardega - Thanks for posting your idea.

In this case, the Salesforce CLI supports these metadata types but it looks like there's a bug in how they're currently handled.  

We're tracking progress on this issue here: #2823 

Could you take a look at that issue and add additional context if appropriate?  Thank you!

**Comment by twardegaSFDC** [comment]
Hi Vivek,

Thank you for a quick follow-up. I will monitor issue #2823 for the
resolution - I will add my additional contacts in that issue. You can close
or attach the issue that I created #2850 as you see fit - seems to be the
same root cause,

Tomek Wardega
Senior Technical Architect | Salesforce
Mobile: 440-334-6327

**Comment by twardegaSFDC** [comment]
Vivek,

I updated issue #2823 - to indicate that the issue affects 2 metadata types:
 DecisionMatrixDefinitionVersion and ExpressionSetDefinitionVersion

I also included a test manifest file on issue #2823:
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
<types>
<members>DAWSON_Case_Type_Mapping_V5</members>
<members>DAWSON_Matter_Queue_Assignment_V1</members>
<members>DAWSON_Event_Code_Mapping_V3</members>
<members>DAWSON_Trial_Location_City_Mapping_V2</members>
<name>DecisionMatrixDefinitionVersion</name>
</types>
<types>
<members>OCC_Dawson_Contact_ES_V1</members>
<members>DAWSONApplicableStagingMatter_V1CLONE1</members>
<members>OCC_Dawson_Matter_ES_V1CLONE0</members>
<members>OCC_Dawson_Legal_File_ES_V1</members>
<members>OCC_Dawson_Practitioner_ES_V1</members>
<name>ExpressionSetDefinitionVersion</name>
</types>
<version>60.0</version>
</Package>
Tomek Wardega
Senior Technical Architect | Salesforce
Mobile: 440-334-6327
